README.md

```diff
@@ -1,2 +1,154 @@
-# holonix
+# Holonix
+
 Holochain app development environment based on Nix.
+
+## Quickstart - Dev shell from default template
+
+hApp developers can use a Nix command to generate a flake that includes Holochain and other required binaries in a dev shell.
+
+```shell
+nix flake init -t github:holochain/holonix
+```
+
+You may be asked to allow substituters to be set to make use of binaries cached on cachix. This is not required to generate the flake, but when invoking the dev shell, Nix can benefit from the cache and download all binaries instead of building them on your machine from scratch.
+
+Inspecting the flake reveals the definition found under [./templates/default/flake.nix](./templates/default/flake.nix).
+
+The command to enter the dev shell and have the defined binaries available for execution is
+
+```shell
+nix develop
+```
+
+With the binary cache configured correctly, most or all binaries will be downloaded. How long this takes depends on your internet connection speed, but in any case the first time the dev shell is invoked will take longer than subsequent times.
+
+After the dev shell has built, Holochain and scaffolding commands are available.
+
+```shell
+[holonix:]$ holochain --version
+holochain x.y
+[holonix:]$ hc-scaffold --version
+holochain_scaffolding_cli x.y
+```
+
+## Updating Holonix flake
+
+At the top of the Nix flake inputs are defined which are the sources of the packages to be included in the dev shell. In the `flake.lock` file the revision of each of the inputs is pinned. If you want to update this pinned revision to the latest state, you can run
+
+```shell
+nix flake update
+```
+
+This will update all inputs to the latest version. In case you just want to update an individual input, append the input identifier to the above command.
+
+```shell
+nix flake update holonix
+```
+
+The next time you enter the dev shell, the updated binaries will be downloaded and provided in the resulting shell. Should you have run `nix flake update` from inside the dev shell, you need to `exit` the shell first and then enter it again with `nix develop`.
+
+[nix flake update in Nix Reference Manual](https://nix.dev/manual/nix/latest/command-ref/new-cli/nix3-flake-update)
+
+## Including other versions of Holochain
+
+By default the Holonix input references the `main` branch of the Holonix repository. Whatever the current revision of that branch at the moment of invoking the dev shell for the first time is will be written to the `flake.lock` file. The `main` branch uses version 0.4 of Holochain and compatible versions of scaffolding etc.
+
+To use other versions of Holochain in the dev shell, the `holonix` input must be modified.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main";
+    ...
+};
+```
+
+For Holochain v0.3, change it to `main-0.3`.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
+    ...
+};
+```
```

Comment:
Could use a `diff` code-block here but that remove syntax highlighting so your choice.

This will give:
```diff
inputs = {
-    holonix.url = "github:holochain/holonix?ref=main";
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
    ...
};
```

```suggestion
To use other versions of Holochain in the dev shell, the `holonix` input must be modified. For Holochain v0.3, change it to `main-0.3`.

```diff
inputs = {
-    holonix.url = "github:holochain/holonix?ref=main";
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
    ...
};
```
```